### DISCUSSION TITLE
Rust based mysql implementation

### DISCUSSION THREAD

**Discussion by somehowchris** [diesel-rs/diesel#3203] [Ideas]
So I'm currently frustrated with the mysql-sys crate thing, so I'm sorry if some things sound unfriendly.

There have been several issues with linking to libmysql-dev which diesel does. From building diesel-cli as a binary, to building problems with musl, static linking issues and much more.

I have gone a bit off the roads and started searching something written in rust. I came across the `mysql` crate https://github.com/blackbeam/rust-mysql-simple And started searching for issues and items in the diesel repo but couldn't fine something. Did someone have an eye on this? Has there been an approach to use something like this?

**Comment by weiznich** [comment]
For the near future we do not plan to implement a pure rust mysql backend in diesel as this is quite a bit of work, would increase the dependency count and would pull in a relatively untested mysql connection implementation (at least compared to the official libmysqlclient  implementation.

That written: Diesel supports writing connection implementation as a third party crate, so someone could step up and provide such a crate. If it turns out to be stable for some time we can talk about merging it into diesel itself. See https://github.com/blackbeam/rust-mysql-simple/discussions/320 for more discussion about this. 

> There have been several issues with linking to libmysql-dev which diesel does. From building diesel-cli as a binary, to building problems with musl, static linking issues and much more.

Another solution for this problem would be to improve the mysqlclient-sys binding crate. Again this would need someone to work on this. Feel free to submit PR's there.
